lib/Doctrine/Common/Collections/ArrayCollection.php

```diff
@@ -31,12 +31,17 @@
  * and may break when we change the internals in the future. If you need to
  * serialize a collection use {@link toArray()} and reconstruct the collection
  * manually.
+ *
+ * @template T
```

Comment:
For the record, I'd prefer if the template would allow specifying a key type as well as the value type: while in many cases you are using the collection as a list with numeric indexes, MongoDB ODM effectively allows using collections as a hash with string indexes. Instead of using `array-key` or `int|string` generically for keys, users should have to consciously specify whether a collection is indexed with numeric or string keys.